### PR TITLE
IGNITE-13459 Document new building process of Ignite C++ (CMake)

### PR DESCRIPTION
**Pull request by ivandasch**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 1]
```diff
@@ -201,21 +201,38 @@ As a result, `ignite-odbc-amd64.msi` and `ignite-odbc-x86.msi` files should appe
...
-Additionally, you will need `GCC`, `G++`, and `make` to build the driver and its dependencies.
+==== Prerequisites
+include::includes/cpp-linux-build-prerequisites.adoc[]
 
-Once all the above mentioned are installed, you can build the Ignite ODBC driver:
+NOTE: JDK is needed currently only for building, odbc driver binary doesn't depend on it.
 
-[source,shell]
+==== Building ODBC driver
+- Create build directory for cmake. We'll refer to this as `{CPP_BUILD_DIR}`
```
Comment (dmagda) [inline docs/_docs/SQL/ODBC/odbc-driver.adoc:210]: I would update the steps this way if you don't mind:

-  Create a build directory for CMake. We'll refer to it as to `{CPP_BUILD_DIR}`.
 - (Optional) Select the prefix of the installation directory which defaults to `/usr/local` if nothing is set. We'll refer to this directory as to `{CPP_INSTALL_DIR}`.
 - Build and install the driver by executing the following commands:

File: docs/_docs/includes/cpp-linux-build-prerequisites.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+The following packages should be installed:
```
Comment (dmagda) [inline docs/_docs/includes/cpp-linux-build-prerequisites.adoc:1]: "Should" or "need to be" (aka. must) installed?

Should imply that something is optional.

File: docs/_docs/quick-start/cpp.adoc [hunk 3]
```diff
@@ -51,30 +44,62 @@ For information about the {cpp} thin client, see link:thin-clients/cpp-thin-clie
...
-=== Running the Thick Client Example
-
-. `cd {IGNITE_HOME}/platforms/cpp/examples`
-. `libtoolize && aclocal && autoheader && automake --add-missing && autoreconf`
-. `./configure`
-. `cd put-get-example`
-. `make`
-. `./ignite-put-get-example`
-
+- Download and unzip the Ignite binary release into a directory. We'll refer to this as `{IGNITE_HOME}`.
```
Comment (dmagda) [inline docs/_docs/quick-start/cpp.adoc:51]: You can remove the "a directory". See how these installation steps are described here: 
https://ignite.apache.org/docs/2.9.0/installation/installing-using-zip#installing-using-zip-archive

File: docs/_docs/quick-start/cpp.adoc [hunk 4]
```diff
@@ -51,30 +44,62 @@ For information about the {cpp} thin client, see link:thin-clients/cpp-thin-clie
...
+- Create build directory for cmake. We'll refer to this as `{CPP_BUILD_DIR}`
```
Comment (dmagda) [inline docs/_docs/quick-start/cpp.adoc:52]: "Create a build directory for CMake"

File: docs/_docs/quick-start/cpp.adoc [hunk 5]
```diff
@@ -51,30 +44,62 @@ For information about the {cpp} thin client, see link:thin-clients/cpp-thin-clie
...
+- Do the folowing:
```
Comment (dmagda) [inline docs/_docs/quick-start/cpp.adoc:53]: "Use the commands below to build and install Ignite C++" - or you can use a shorter version. The point is that I would say that the reader needs to "use the commands" to do X.

File: docs/_docs/quick-start/cpp.adoc [hunk 6]
```diff
@@ -51,30 +44,62 @@ For information about the {cpp} thin client, see link:thin-clients/cpp-thin-clie
...
+cmake3 -DCMAKE_BUILD_TYPE=Release -DWITH_ODBC=ON -DWITH_THIN_CLIENT=ON ${IGNITE_HOME}/platforms/cpp 
+make 
+sudo make install
+----
+
+--
+
+
+=== Building and running the Thick Client Example
+- Create build directory for cmake. We'll refer to this as `{CPP_EXAMPLES_BUILD_DIR}`
```
Comment (dmagda) [inline docs/_docs/quick-start/cpp.adoc:79]: Same feedback as above, you need to put articles in some places. Just adjust the way you would adjust the instructions above.

**Review by dmagda — CHANGES_REQUESTED** [review]

File: docs/_docs/SQL/ODBC/odbc-driver.adoc [hunk 7]
```diff
@@ -201,21 +201,38 @@ As a result, `ignite-odbc-amd64.msi` and `ignite-odbc-x86.msi` files should appe
 
 On a Linux-based operating system, you will need to install an ODBC Driver Manager of your choice to be able to build and use the Ignite ODBC Driver. The ODBC Driver has been tested with link:http://www.unixodbc.org[UnixODBC].
 
-Additionally, you will need `GCC`, `G++`, and `make` to build the driver and its dependencies.
+==== Prerequisites
+include::includes/cpp-linux-build-prerequisites.adoc[]
 
-Once all the above mentioned are installed, you can build the Ignite ODBC driver:
+NOTE: JDK is needed currently only for building, odbc driver binary doesn't depend on it.
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply docs/_docs/SQL/ODBC/odbc-driver.adoc]: Fixed.

File: docs/_docs/includes/cpp-linux-build-prerequisites.adoc (hunk 2 above)
Comment (ivandasch) [reply docs/_docs/includes/cpp-linux-build-prerequisites.adoc]: Fixed.

File: docs/_docs/quick-start/cpp.adoc (hunk 3 above)
Comment (ivandasch) [reply docs/_docs/quick-start/cpp.adoc]: Fixed.

File: docs/_docs/quick-start/cpp.adoc (hunk 4 above)
Comment (ivandasch) [reply docs/_docs/quick-start/cpp.adoc]: Fixed

File: docs/_docs/quick-start/cpp.adoc (hunk 6 above)
Comment (ivandasch) [reply docs/_docs/quick-start/cpp.adoc]: Fixed

File: docs/_docs/SQL/ODBC/odbc-driver.adoc (hunk 1 above)
Comment (ivandasch) [reply docs/_docs/SQL/ODBC/odbc-driver.adoc]: Fixed

File: docs/_docs/includes/cpp-linux-build-prerequisites.adoc (hunk 2 above)
Comment (ivandasch) [reply docs/_docs/includes/cpp-linux-build-prerequisites.adoc]: Fixed.

**Pull request merged**